Detectors/MUON/MCH/Calibration/src/BadChannelCalibrator.cxx

```diff
@@ -61,14 +71,13 @@ bool BadChannelCalibrator::hasEnoughData(const Slot& slot) const
   const int requiredChannels = static_cast<int>(BadChannelCalibratorParam::Instance().minRequiredCalibratedFraction * nofChannels);
 
   auto nofCalibrated = std::count_if(pedData->cbegin(), pedData->cend(),
-                                     [&](const PedestalChannel& c) { return c.mEntries > minNofEntries; });
+                                     [&](const PedestalChannel& c) { return (c.mValid && (c.mEntries > minNofEntries)); });
```

Comment:
actually, instead of testing for `mValid` (see also the earlier comment about whether or not we really need this extra boolean) here and in line 103, the skipping of the non existing channel could be baked in the iterator directly.

Reply:
I am not sure to see how to implement this in the iterator... I have replaced the `mValid` member by a `isValid()` function that actually checks the value of the `dsChannelId`.

Reply:
for the iterator, see https://github.com/AliceMCH/AliceO2/pull/34